pkg/collector/corechecks/snmp/snmp_test.go

```diff
@@ -2043,16 +2053,17 @@ metric_tags:
 
 func TestDeviceIDAsHostname(t *testing.T) {
 	cache.Cache.Delete(cache.BuildAgentKey("hostname")) // clean existing hostname cache
-	coreconfig.Datadog.Set("hostname", "test-hostname")
-	coreconfig.Datadog.Set("tags", []string{"agent_tag1:val1", "agent_tag2:val2"})
 
 	checkconfig.SetConfdPathAndCleanProfiles()
 	sess := session.CreateMockSession()
 	sessionFactory := func(*checkconfig.CheckConfig) (session.Session, error) {
 		return sess, nil
 	}
 	chk := Check{sessionFactory: sessionFactory}
-	aggregator.InitAndStartAgentDemultiplexerTest(demuxOpts(), "")
+	forwarder := fxutil.Test[defaultforwarder.Component](t, defaultforwarder.MockModule, config.MockModule)
+	coreconfig.Datadog.Set("hostname", "test-hostname")
```

Comment:
It'd be better if we can try to avoid making calls to globals in our tests. I know this is beyond the scope of this PR, and is something that needs to be addressed in multiple places, just mentioning it here for reference.

Reply:
This code was moved and I didn't want to cleanup all the tests as the PR is quite big.